parse/sql/sqlanalyzer/typing/visitor.go

```diff
@@ -26,103 +28,85 @@ type typeVisitor struct {
 var _ tree.AstVisitor = &typeVisitor{}
 
 // evalFunc is a function that allows modifying an evaluation context.
-type evalFunc func(e *evaluationContext) error
+type evalFunc func(e *evaluationContext)
 
 // BEGIN evalFunc
 
 func (t *typeVisitor) VisitCTE(p0 *tree.CTE) any {
-	return evalFunc(func(e *evaluationContext) error {
-		relation, err := p0.Select.Accept(t).(returnFunc)(e)
-		if err != nil {
-			return err
-		}
+	return evalFunc(func(e *evaluationContext) {
+		relation := p0.Select.Accept(t).(returnFunc)(e)
 
 		_, ok := t.commonTables[p0.Table]
 		if ok {
-			return fmt.Errorf("common table expression conflicts with existing table %s", p0.Table)
+			t.options.ErrorListener.NodeErr(p0.GetNode(), parseTypes.ParseErrorTypeSemantic, fmt.Errorf("common table expression conflicts with existing table %s", p0.Table))
```

Comment:
seems nbd, but `t.options.ErrorListener` feels a bit off to me, semantically.  It's option, should be used to overwrite your default `t.ErrorListener` .

Reply:
The options struct sets a default if none are set. Will make it more obvious